### ISSUE TITLE
Asking to install FUSE while I have already done that #mac 

### ISSUE THREAD

**Issue by azataiot** [s3fs-fuse/s3fs-fuse#1618]
### Additional Information
_The following information is very important in order to help us to help you. Omission of the following details may delay your support request or receive no attention at all._
_Keep in mind that the commands we provide to retrieve information are oriented to GNU/Linux Distributions, so you could need to use others if you use s3fs on macOS or BSD_

MacBook Air (13-inch, 2017) Core Intel Core i5
Mac Os big sur version 11.2.3

--- 
❯ brew install --cask osxfuse
Warning: Cask 'osxfuse' is already installed.

To re-install osxfuse, run:
  brew reinstall osxfuse

---
❯ brew install s3fs
Error: s3fs has been disabled because it requires FUSE!
----

#### Version of s3fs being used (s3fs --version)
_example: 1.00_

#### Version of fuse being used (pkg-config --modversion fuse, rpm -qi fuse, dpkg -s fuse)
_example: 2.9.4_

#### Kernel information (uname -r)
_command result: uname -r_

#### GNU/Linux Distribution, if applicable (cat /etc/os-release)
_command result: cat /etc/os-release_

#### s3fs command line used, if applicable
```
```
#### /etc/fstab entry, if applicable
```
```
#### s3fs syslog messages (grep s3fs /var/log/syslog, journalctl | grep s3fs, or s3fs outputs)
_if you execute s3fs with dbglevel, curldbg option, you can get detail debug messages_
```
```

**Comment by gaul** [comment]
Homebrew removed support for FUSE and all related formula in Homebrew/homebrew-core#64491.  I believe that there was a lack of maintainership of macfuse which prompted this.  You will need to compile s3fs yourself or follow up with the Homebrew maintainers.

**Comment by gaul** [comment]
Related to #1632.

**Comment by gaul** [comment]
Related: GoogleCloudPlatform/gcsfuse#514

**Comment by gaul** [comment]
Could you try this external Homebrew cask:

https://github.com/gromgit/homebrew-fuse/

If it works can you update README.md?

**Comment by dloeckx** [comment]
This seems to work indeed (on Apple Intel):

First do `brew install --cask macfuse` (this was requested by s3fs-mac), and next `brew install gromgit/fuse/s3fs-mac`. If somebody can confirm, I will update the README.md.

**Comment by jamietownsend** [comment]
Wow - perfect timing. I can confirm that the approach above from @dloeckx correctly installs s3fs. Thanks heaps!

**Comment by mcoliver** [comment]
> brew install gromgit/fuse/s3fs-mac

That path is interesting but I don't know gromgit and their formula is serving up compiled releases of s3fs from their own repo.  I don't think that is necessary and is a potential security concern.  

[https://github.com/gromgit/homebrew-fuse/blob/5637fc5f093756d36ee999bc490a866a469051e4/Formula/s3fs-mac.rb#L12](https://github.com/gromgit/homebrew-fuse/blob/5637fc5f093756d36ee999bc490a866a469051e4/Formula/s3fs-mac.rb#L12)

So perhaps brew formula could be hosted out of the main s3fs repo?  precompiled releases also, or just let brew compile from source and don't serve up any bottles.

**Comment by gaul** [comment]
We would be happy to host the canonical formula but precompiled releases are probably too much work.  Forcing users to compile s3fs is inconvenient but it is a small project with few dependencies so it should not be a big issue.  Could you submit a PR with both the formula and the updated instructions?

**Comment by mmmdamin** [comment]
thank you @dloeckx
it works

**Comment by terryzwt** [comment]
@dloeckx Works for me.
Also resolve my pending problem for installing ntfs-3g.
```
brew install --cask macfuse
brew install gromgit/fuse/ntfs-3g
```

**Comment by spolischook** [comment]
@XDavidT 
```
brew install gromgit/fuse/s3fs-mac
```

**Comment by XDavidT** [comment]
> @XDavidT
> 
> ```
> brew install gromgit/fuse/s3fs-mac
> ```

Great, now no error when mount, but the mount is stuck, I can't do anything and my filesystem is always busy. help

**Comment by spolischook** [comment]
not my issue. For me it works. Try reinstall

**Comment by gaul** [comment]
Does #1817 resolve this issue?

**Comment by rajivravio** [comment]
> This seems to work indeed (on Apple Intel):
> 
> First do `brew install --cask macfuse` (this was requested by s3fs-mac), and next `brew install gromgit/fuse/s3fs-mac`. If somebody can confirm, I will update the README.md.

Yup, this works, I have tested it twice. Currently osxfuse still installs through brew but it doesn't seem to work and still requires macfuse.

**Comment by jrd2017** [comment]
I was able to install the macfuse by downloading the file from the website and installing it from te DMG file.  To mount the External Storage, S3 bucket in my case, I downloaded the use rclone, following this video.  [https://www.youtube.com/watch?v=29EPlJpeuhM&t=7s
]


I hope this helps you!  Cheers

**Comment by vicary** [comment]
> This seems to work indeed (on Apple Intel):
> 
> 
> 
> First do `brew install --cask macfuse` (this was requested by s3fs-mac), and next `brew install gromgit/fuse/s3fs-mac`. If somebody can confirm, I will update the README.md.

Tried installing on Apple M1, it hangs indefinitely and I can't even gracefully reboot. I guess it is blocking at some I/O syscalls.

**Comment by Provoo** [comment]
> > @XDavidT
> > ```
> > brew install gromgit/fuse/s3fs-mac
> > ```
> 
> Great, now no error when mount, but the mount is stuck, I can't do anything and my filesystem is always busy. help

I have the same problem any updates?

**Comment by alfiedotwtf** [comment]
Still happening here too on an M2

**Comment by alfiedotwtf** [comment]
Ok, I think I found something interesting - the m2 hanging problem seems to be fixed when you use `-f`, which keeps it in the foreground. People on m2: give it a go and see if it works!

**Comment by ggtakec** [comment]
https://github.com/s3fs-fuse/s3fs-fuse/issues/1632#issuecomment-1484077046
